src/custom-lists/background/storage.ts

```diff
@@ -0,0 +1,191 @@
+        super(storageManager)
+
+        this.storageManager.registerCollection(COLLECTION_NAME, {
+            // different version for adding a new table.
+            version: new Date(2018, 6, 12),
+            fields: {
+                id: { type: 'string', pk: true },
+                name: { type: 'text' },
+                isDeletable: { type: 'binary' },
+                isNestable: { type: 'binary' },
+                createdAt: { type: 'datetime' },
+            },
+            indices: [
+                { field: 'id', pk: true, autoInc: true },
+                { field: 'name', unique: true },
+                { field: 'isDeletable' },
+                { field: 'isNestable' },
+                { field: 'createdAt' },
+            ],
+        })
+
+        this.storageManager.registerCollection(PAGE_LIST_ENTRY, {
+            // different version for adding a new table.
+            version: new Date(2018, 6, 12),
+            fields: {
+                listId: { type: 'string' },
+                pageUrl: { type: 'string' },
+                fullUrl: { type: 'string' },
+                createdAt: { type: 'datetime' },
+            },
+            indices: [
+                { field: ['listId', 'pageUrl'], pk: true },
+                { field: 'listId' },
+                { field: 'pageUrl' },
+            ],
+        })
+    }
+
+    // Return all the list in the DB
+    //  TODO: Use pagination if required and decide correct types
+    async fetchAllList({ query = {}, opts = {} }: { query?: any; opts?: any }) {
+        const x = await this.storageManager.findAll(COLLECTION_NAME, query, opts)
+        return this.changeListsbeforeSending(x, [])
+    }
+
+    // Takes the list as they come from Db and does some pre-processing before sending.
+    async changeListsbeforeSending(lists: object[], pageEnteries: object[]) {
+        const mappedLists = lists.map((list: ListObject) => {
+            const page = pageEnteries.find(({ listId }: PageObject) => listId === list.id)
+            delete list["_name_terms"]
+            return {
+                ...list,
+                // Set an empty pages array for manupulating in redux store.
+                pages: [],
+                active: Boolean(page),
+            }
+        })
+        return mappedLists
+    }
+
+    async fetchListById(id: Number) {
+        const list = await this.storageManager.findObject(COLLECTION_NAME, { id })
+        const pages = await this.storageManager.findAll(PAGE_LIST_ENTRY, { listId: list.id })
+        delete list["_name_terms"]
+        return {
+            ...list,
+            pages: pages.map((page: PageObject) => page.fullUrl),
+        }
+    }
+
+    // Return all the pages associated with a list.
+    async fetchListPages(listId: Number) {
+        return await this.storageManager.findAll(PAGE_LIST_ENTRY, { listId })
+    }
+
+    // Returns all the lists containing a certain page.
+    async getListAssocPage({ url }: { url: string }) {
```

Comment:
Maybe this would be better called `fetchPageLists` (to be more similar to `fetchListPages`, which provides a similar operation on this relationship)? 

Another semi-related point here is to be consistent with operation verbs like `get` and `fetch`. I know throughout the code-base it's really messed up with this sort of thing, but good to try and be consistent with new code. I personally lean towards the `fetch` verb as, to me, it indicates some async operation (`get` makes me think of a standard sync class getter in standard JS/TS)